pkg/worker/verifier/producer_worker.go

```diff
@@ -121,10 +121,15 @@ func (pw *ProducerWorker) newRecord(producerId int, sequence int64) *kgo.Record
 		pw.Status.AbortedTransactionMessages += 1
 	}
 
-	payload := pw.config.valueGenerator.Generate()
-	if payload == nil {
+	var payload []byte
+	isTombstone := rand.Float64() < pw.config.valueGenerator.TombstoneProbability
+	if isTombstone {
+		payload = nil
 		pw.Status.TombstonesProduced += 1
+	} else {
+		payload = make([]byte, pw.config.messageSize)
```

Comment:
Presumably the values weren't important anyway since compaction tests only validate keys anyway, is that right?

Reply:
They are valuable, as there is latest key-value validation in the verifier now.

I am grappling with Go to figure out the best way to safely encode something like `value-offset` in the payload.

Reply:
Though, encoding `value-offset` in the message will possibly result in message sizes > `pw.config.messageSize`.

So, do we guarantee the correct message size, or do we ensure correct validation for key-value pairs?